slothy/targets/arm_v7m/arch_v7m.py

```diff
@@ -297,6 +297,97 @@ def end(self, other, indentation=0):
         yield f'{indent}cmp {other["cnt"]}, {other["end"]}'
         yield f'{indent}bne {lbl_start}'
 
+
+class BranchLoop(Loop):
```

Comment:
```suggestion
class BranchLoop(Loop):
    """
    More general loop type that just considers the branch instruction as part of the boundary.
    This can help to improve performance as the instructions that belong to handling the loop can be considered by SLOTHY aswell. 
    
    Note: This loop type is still rather experimental. It has a lot of logics inside as it needs to be able to "understand" a variety of different ways to express loops, e.g., how counters get incremented, how registers marking the end of the loop need to be modified in case of software pipelining etc. Currently, this type covers the three other types we offer above, namely `SubsLoop`, `CmpLoop`, and `VmovCmpLoop`. 

    For examples, we refer to the classes `SubsLoop`, `CmpLoop`, and `VmovCmpLoop`. 
 """
```